src/test/java/com/google/sps/servlets/StudyServletTest.java

```diff
@@ -1 +1,305 @@
+    cards.add(cardBInDatastore);
+    cards.add(cardCInDatastore);
+
+    servlet.doGet(mockRequest, mockResponse);
+    String response = responseWriter.toString();
+    String expectedResponse =
+        "["
+            + "[{\"quizWord\":\"translatedTest2\", \"cardKey\":\"agR0ZXN0chwLEgZGb2xkZXIiBnRlc3RJRAwLEgRDYXJkGAIM\","
+            + "\"possibleResponses\":[\"test2\"],\"correctAnswer\":\"test2\"}],"
+            + "[{\"quizWord\":\"translatedTest\",\"cardKey\":\"agR0ZXN0chwLEgZGb2xkZXIiBnRlc3RJRAwLEgRDYXJkGAEM\","
+            + "\"possibleResponses\":[\"test\"],\"correctAnswer\":\"test\"}],"
+            + "[{\"quizWord\":\"translatedTest3\",\"cardKey\":\"agR0ZXN0chwLEgZGb2xkZXIiBnRlc3RJRAwLEgRDYXJkGAMM\","
+            + "\"possibleResponses\":[\"test3\"],\"correctAnswer\":\"test3\"}]"
+            + "]";
+
+    assertTrue(compareJson(response, expectedResponse));
+  }
+
+  @Test
+  public void folderHasNoCards() throws Exception {
+    List<Card> noCardsInDatastore = new ArrayList<>();
+
+    Entity folder = new Entity("Folder", "testID");
+    String folderKey = KeyFactory.keyToString(folder.getKey());
+
+    when(mockRequest.getParameter("folderKey")).thenReturn(folderKey);
+    servlet.doGet(mockRequest, mockResponse);
+    String response = responseWriter.toString();
+    String expectedResponse = "[]";
+
+    assertTrue(compareJson(response, expectedResponse));
+  }
+
+  @Test
+  public void userNotLoggedIn() throws Exception {
+    helper.setEnvIsLoggedIn(false);
+    List<Card> noCardsQueried = new ArrayList<>();
+
+    servlet.doGet(mockRequest, mockResponse);
+    String response = responseWriter.toString();
+    String expectedResponse = "";
+
+    assertTrue(compareJson(response, expectedResponse));
+  }
+
+  @Test
+  public void userAnswersCorrectly() throws Exception {
+    // The time gap between when the user created the card and when its being tested is 45 hours
+    cardA =
+        new Card.Builder()
+            .setFamiliarityScore(.5)
+            .setTimeTested(0L)
+            .setImageBlobKey("null")
+            .setRawText("test")
+            .setTextTranslated("translatedTest")
+            .build();
+
+    Entity folder = new Entity("Folder", "testID");
+    String folderKey = KeyFactory.keyToString(folder.getKey());
+
+    Card cardInDatastore = storeCardInDatastore(cardA, datastore, folderKey);
+    String cardKey = cardInDatastore.getCardKey();
+
+    when(mockRequest.getParameter("answeredCorrectly")).thenReturn("true");
+    when(mockRequest.getParameter("cardKey")).thenReturn(cardKey);
+
+    servlet.doPost(mockRequest, mockResponse);
+
+    Entity updatedCard = datastore.get(KeyFactory.stringToKey(cardKey));
+    Card cardAUpdated = new Card(updatedCard);
+    Double response = cardAUpdated.getFamiliarityScore();
+    long time = cardAUpdated.getTimeTested();
+    Double expectedResponse = (Double) 2.142857142;
+
+    assertTrue(compareDoubles(response, expectedResponse));
```

Comment:
Hardcode the expected response into compareDoubles. No need for another variable because the literal is so short.

Reply:
done.